### PR TITLE
Improve maintainability of theme json class tests.

### PR DESCRIPTION
**Pull request by tellthemachines**
## What?

Partially addresses #60842.

This PR syncs https://github.com/WordPress/wordpress-develop/pull/6734 to Gutenberg. I also adjusted a couple tests that don't exist in core yet to not output base layout styles.

## Testing Instructions

1. Run the PHP tests and check that everything passes.

### CODE REVIEW

File: phpunit/class-wp-theme-json-test.php [hunk 1]
```diff
@@ -1099,12 +1044,12 @@ public function test_get_stylesheet_with_deprecated_feature_level_selectors() {
 			)
 		);
```
Comment (tellthemachines) [inline phpunit/class-wp-theme-json-test.php:1030]: If I'm reading the comments above this test and the next one correctly, we should be able to update them to use core blocks now?

Comment (aaronrobertshaw) [reply phpunit/class-wp-theme-json-test.php:1030]: That's my understanding. We should be able to update these tests to core blocks and remove the filter registering the test block from the phpunit bootstrap.

I can look a little deeper on this tomorrow if that helps?

Comment (tellthemachines) [reply phpunit/class-wp-theme-json-test.php:1030]: Thanks for confirming, I'll give it a go and see if it works!

Comment (tellthemachines) [reply phpunit/class-wp-theme-json-test.php:1030]: Does the "deprecated" in the test name refer to defining selectors with `__experimentalSelector` inside the `supports` array (like Calendar or Button do) as opposed to outside (like Image does)?

Comment (andrewserong) [reply phpunit/class-wp-theme-json-test.php:1030]: I believe so, it looks like this test was renamed from `test_get_stylesheet_with_block_support_feature_level_selectors` in https://github.com/WordPress/gutenberg/pull/46496/ which stabilised that feature and moved it outside of `supports`.

Comment (aaronrobertshaw) [reply phpunit/class-wp-theme-json-test.php:1030]: The selectors API feature got merged by a good samaritan before it was actually ready so following threads like #46486 will be tricky as the individual PRs only contain part of the "story".

To answer the question, yes, the deprecated `__experimentalSelector` properties are what the test was about as we have to maintain backwards compatibility for them unfortunately.

Comment (tellthemachines) [reply phpunit/class-wp-theme-json-test.php:1030]: Thanks folks, all done! I trimmed the properties output to match what each block supports, and that also makes the strings a more manageable size.

File: phpunit/class-wp-theme-json-test.php [hunk 2]
```diff
@@ -4801,36 +4689,17 @@ public function test_get_top_level_background_image_styles() {
...
-		$this->assertSame( $expected_styles, $theme_json->get_stylesheet(), 'Styles returned from "::get_stylesheet()" with top-level background styles type does not match expectations' );
+		$body_node = array(
+			'path'     => array( 'styles' ),
+			'selector' => 'body',
+		);
+
+		$expected_styles = "html{min-height: calc(100% - var(--wp-admin--admin-bar--height, 0px));}:root :where(body){background-image: url('http://example.org/image.png');background-position: center center;background-repeat: no-repeat;background-size: contain;}";
+		$this->assertSame( $expected_styles, $theme_json->get_styles_for_block( $body_node ), 'Styles returned from "::get_stylesheet()" with top-level background styles type does not match expectations' );
```
Comment (andrewserong) [inline phpunit/class-wp-theme-json-test.php]:
```suggestion
		$this->assertSame( $expected_styles, $theme_json->get_styles_for_block( $body_node ), 'Styles returned from "::get_styles_for_block()" with top-level background styles type does not match expectations' );
```

File: phpunit/class-wp-theme-json-test.php [hunk 3]
```diff
@@ -4846,8 +4715,8 @@ public function test_get_top_level_background_image_styles() {
...
-		$this->assertSame( $expected_styles, $theme_json->get_stylesheet(), 'Styles returned from "::get_stylesheet()" with top-level background image as string type does not match expectations' );
+		$expected_styles = "html{min-height: calc(100% - var(--wp-admin--admin-bar--height, 0px));}:root :where(body){background-image: url('http://example.org/image.png');background-position: center center;background-repeat: no-repeat;background-size: contain;}";
+		$this->assertSame( $expected_styles, $theme_json->get_styles_for_block( $body_node ), 'Styles returned from "::get_stylesheet()" with top-level background image as string type does not match expectations' );
```
Comment (andrewserong) [inline phpunit/class-wp-theme-json-test.php]:
```suggestion
		$this->assertSame( $expected_styles, $theme_json->get_styles_for_block( $body_node ), 'Styles returned from "::get_styles_for_block()" with top-level background image as string type does not match expectations' );
```

**Review by andrewserong — APPROVED** [review]
Looking good — very cool that real core blocks could be used for both cases of the deprecated and proper selectors, too, that helps simplify things a fair bit.

Just left a couple of tiny nit comments in the strings that are output on test failures, but other than that this LGTM! 🚀

File: phpunit/class-wp-theme-json-test.php (hunk 3 above)
Comment (tellthemachines) [reply phpunit/class-wp-theme-json-test.php]: Oh, well spotted, thanks!

**Pull request merged**